docs/src/tutorials/custom_impurity_density_pn_junction_lit.jl

```diff
@@ -0,0 +1,143 @@
+# # Advanced Example: Custom Impurity Profile
+
+# A very important input to the simulation is the impurity profile of the semiconductor.
+# It influences the electric potential and, thus, the electric field, capacitances,
+# drift paths and also the induced waveforms.
+
+# Arbitrary impurity density profiles can be defined and assigned to the semiconductor of the detector.
+# Also have a look into the manual: [Impurity Densities](@ref) and [Custom Impurity Density](@ref).
+
+# As an example, we define the impurity profile of a simple [p-n junction](https://en.wikipedia.org/wiki/P%E2%80%93n_junction)
+# and calculate the electric potential and depleted region and reproduce the 
+# typical [plot](https://en.wikipedia.org/wiki/P%E2%80%93n_junction#/media/File:Pn-junction-equilibrium-graphs.png)
+# shown in various books explaining p-n junctions.
+
+# For this, we use one of the example detectors: the infinite parallel plate capacitor in cartesian coordinates.
+# The plates span over `y` and `z`. It is made infinite by defining the world smaller than the contacts
+# and using reflecting boundaries in `y` and `z`.
+# Let's get started by loading the packages and the detector and have a look at its geometry:
+
+using Plots
+using SolidStateDetectors
+using Unitful
+
+sim = Simulation{Float32}(SSD_examples[:InfiniteParallelPlateCapacitor])
+plot(sim.detector)
+
+# ## Define the Impurity Density
+
+# Now, we define the model for impurity density of the p-n junction.
+# In order to do so, we need two define two things.
+
+# 1. We have to define a `struct` which has to be 
+# subtype of `SolidStateDetectors.AbstractImpurityDensity{T}`.
+# We define the model with two parameters in order to be able to vary the density level 
+# on the p-type and n-type regions independently. 
+# More parameters could be added. E.g., the position of the p-n junction.
+
+struct PNJunctionImpurityDensity{T} <: SolidStateDetectors.AbstractImpurityDensity{T} 
+    p_type_density::T
+    n_type_density::T
+end
+
+# 2. We have to define a method for `SolidStateDetectors.get_impurity_density` for our 
+# just defined impurity density. 
+# It has to take two arguments. An instance of our model and a point.
+# Since the infinite parallel plate capacitor is configured to be calculated on a cartesian grid,
+# the point has to be a `CartesianPoint`.
```

Comment:
If I weren't used to SSD, I would read this as: for every new detector, you need to define a new impurity density profile. So what we are doing here sounds not applicable/usable for cylindrical grids.

However, (and this is also something we do for the pre-defined impurity density profiles), we can just do
```julia
function SolidStateDetectors.get_impurity_density(cdm::PNJunctionImpurityDensity{T},  pt::CylindricalPoint{T})::T where {T}
    SolidStateDetectors.get_impurity_density(cdm, CartesianPoint(pt))
end
```
Can we somehow add that here?

Ànd a general comment: Is there any reason, why we don't generally go for:
```julia
function SolidStateDetectors.get_impurity_density(
    cdm::PNJunctionImpurityDensity{T}, 
    pt::AbstractCoordinatePoint{T}
)::T where {T}
    cpt = CartesianPoint(pt) # converts it to CartesianPoint, does nothing if it is a CartesianPoint
    x = cpt[1] # In this example, we only need the `x` coordinate of the point.
    if x > 0
        -cdm.p_type_density # p-type region -> electrons are fixed -> negative charge 
    else 
        cdm.n_type_density  # n-type region -> holes are fixed -> positive charge
    end
end
```

`